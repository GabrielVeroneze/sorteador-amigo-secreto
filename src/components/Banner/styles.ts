import styled from 'styled-components'
import { device } from '@/styles/Breakpoints'

export const Container = styled.header`
    align-items: center;
    background-color: ${({ theme }) => theme.cores.primaria};
    display: flex;
    flex-direction: column;
    height: 384px;
    justify-content: space-between;
    padding: 2rem ${({ theme }) => theme.paddings.lateral.pequeno};

    @media ${device.lg} {
        flex-direction: row;
        height: 454px;
        justify-content: center;
        padding: 3.5rem;
    }
`

export const Logo = styled.h1`
    background-image: url('images/logo-vertical.svg');
    background-repeat: no-repeat;
    background-size: cover;
    height: 157px;
    width: 186px;

    @media ${device.lg} {
        background-image: url('images/logo-horizontal.svg');
        height: 117px;
        width: 351px;
    }
`

export const Imagem = styled.picture`
    z-index: 1;

    @media ${device.lg} {
        align-self: flex-end;
    }
`
