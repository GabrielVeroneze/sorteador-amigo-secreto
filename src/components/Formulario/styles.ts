import styled from 'styled-components'

export const Campo = styled.form`
    border-radius: 37.5px;
    box-shadow: 4px 4px 0px 0px #000000;
    color: ${({ theme }) => theme.cores.texto.primaria};
    display: flex;
    font-size: 1rem;
    height: 82px;
    position: relative;
`

export const Icone = styled.div`
    color: #0000004d;
    left: 52px;
    position: absolute;
    top: 28px;
`

export const Input = styled.input`
    border: 2px solid #000000;
    border-radius: 37.5px 0px 0px 37.5px;
    flex-grow: 1;
    outline: none;
    padding: 1.125rem 2.25rem 1.125rem 6.625rem;

    &::placeholder {
        color: #0000004d;
    }
`

export const Botao = styled.button`
    background-color: ${({ theme }) => theme.cores.fundo.secundario};
    border: 2px solid #000000;
    border-radius: 0px 37.5px 37.5px 0px;
    width: 227px;
`
