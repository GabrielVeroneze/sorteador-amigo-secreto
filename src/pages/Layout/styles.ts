import styled from 'styled-components'

export const Principal = styled.main`
    background-color: ${({ theme }) => theme.cores.fundo.primaria};
    border: 1px solid #000000;
    border-radius: 48px 48px 0px 0px;
    flex-grow: 1;
`
