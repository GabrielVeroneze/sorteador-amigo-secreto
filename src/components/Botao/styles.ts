import styled from 'styled-components'
import { device } from '@/styles/Breakpoints'

export const Button = styled.button`
    background-color: ${({ theme }) => theme.cores.secundaria};
    border: 1px solid #000000;
    border-radius: 37.5px;
    box-shadow: 2px 2px 0px 0px #000000;
    color: ${({ theme }) => theme.cores.texto.secundaria};
    font-size: 1rem;
    font-weight: 600;
    height: 66px;
    letter-spacing: 0.25px;
    padding: 0 1.5rem;
    width: 155px;

    &:disabled {
        cursor: not-allowed;
        opacity: 0.6;
    }

    svg {
        display: none;
    }

    @media ${device.lg} {
        align-items: center;
        border: 2px solid #000000;
        box-shadow: 4px 4px 0px 0px #000000;
        display: flex;
        font-size: 1.25rem;
        gap: 1.625rem;
        height: 80px;
        padding: 0 2.5rem;
        width: auto;

        svg {
            display: inline-block;
            font-size: 2.5rem;
        }
    }
`
