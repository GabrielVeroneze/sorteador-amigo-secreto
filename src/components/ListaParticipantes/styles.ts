import styled from 'styled-components'
import { device } from '@/styles/Breakpoints'

export const Lista = styled.ul`
    align-items: center;
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    margin: 1.125rem 0;

    @media ${device.lg} {
        margin: 1.875rem 0 1rem;
    }
`

export const Participante = styled.li`
    color: ${({ theme }) => theme.cores.texto.primaria};
    font-size: 1rem;
    line-height: normal;

    @media ${device.lg} {
        font-size: 1.125rem;
    }
`
