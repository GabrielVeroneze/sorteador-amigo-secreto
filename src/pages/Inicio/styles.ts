import styled from 'styled-components'
import { device } from '@/styles/Breakpoints'

export const Wrapper = styled.div`
    align-items: center;
    align-self: self-end;
    display: flex;
    flex-direction: column;
    gap: 1.5rem;

    @media ${device.lg} {
        flex-direction: row;
        justify-content: space-between;
    }
`
