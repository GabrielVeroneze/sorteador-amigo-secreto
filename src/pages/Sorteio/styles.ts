import styled from 'styled-components'
import { device } from '@/styles/Breakpoints'

export const MenuOpcoes = styled.select`
    border: 1px solid #000000;
    border-radius: 32px;
    box-shadow: 2px 2px 0px 0px #000000;
    color: ${({ theme }) => theme.cores.texto.primaria};
    font-family: ${({ theme }) => theme.fontes.familia.primaria};
    font-size: 1rem;
    height: 50px;
    line-height: normal;
    outline: none;
    padding: 0 1.75rem;
    width: 100%;

    @media ${device.lg} {
        border: 2px solid #000000;
        box-shadow: 4px 4px 0px 0px #000000;
        height: 75px;
        padding: 0 2rem;
    }
`
