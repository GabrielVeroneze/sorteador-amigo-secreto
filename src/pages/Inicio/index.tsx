import { useListaDeParticipantes } from '@/hooks/useListaDeParticipantes'
import { Lista, Participante } from './styles'
import Formulario from '@/components/Formulario'
import Botao from '@/components/Botao'

const Inicio = () => {
    const { listaDeParticipantes } = useListaDeParticipantes()

    return (
        <Formulario />
        <>
            <Formulario />
            <Lista>
                {listaDeParticipantes.map((participante, index) => (
                    <Participante key={index}>
                        {participante}
                    </Participante>
                ))}
            </Lista>
        </>
    )
}

export default Inicio
