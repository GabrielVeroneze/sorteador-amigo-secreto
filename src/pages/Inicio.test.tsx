import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { RecoilRoot } from 'recoil'
import { theme } from '@/styles/Theme'
import Inicio from '@/pages/Inicio'

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavegacao
}))

describe('a página de configuração', () => {
    test('deve ser renderizada corretamente', () => {
        const { container } = render(
            <ThemeProvider theme={theme}>
                <RecoilRoot>
                    <Inicio />
                </RecoilRoot>
            </ThemeProvider>
        )

        expect(container).toMatchSnapshot()
    })
})
