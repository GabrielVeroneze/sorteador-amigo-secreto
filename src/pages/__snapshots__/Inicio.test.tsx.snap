// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <h2
    class="sc-bdDqHD hTZBMr"
  >
    Vamos começar!
  </h2>
  <form
    class="sc-gtcySc cnlRmv"
  >
    <div
      class="sc-dkXqGt eAJEYD"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PersonAddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <input
      class="sc-hKVqoC jArpWw"
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      class="sc-eCQicT dFdYml"
      disabled=""
    >
      Adicionar
    </button>
  </form>
  <ul
    class="sc-gKkibJ bNYHlP"
  />
  <div
    class="sc-fuzrqj cXeCqa"
  >
    <button
      class="sc-qdOTD deggxj"
      disabled=""
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PlayCircleOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m10 16.5 6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
      Iniciar brincadeira!
    </button>
    <picture>
      <source
        media="(min-width: 1024px)"
      />
      <img
        alt="Sacolas de compras"
      />
    </picture>
  </div>
</div>
`;
